src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -2969,77 +2958,49 @@ private Keyspace getValidKeyspace(String keyspaceName)
      * Remove the snapshot with the given name from the given keyspaces.
      * If no tag is specified we will remove all snapshots.
      */
-    public void clearSnapshot(String tag, String... keyspaceNames)
+    public void clearSnapshot(String tag, String... keyspaceNames) throws IOException
     {
         clearSnapshot(Collections.emptyMap(), tag, keyspaceNames);
     }
 
     public void clearSnapshot(Map<String, Object> options, String tag, String... keyspaceNames)
     {
-        if (tag == null)
-            tag = "";
-
         if (options == null)
             options = Collections.emptyMap();
 
-        Set<String> keyspaces = new HashSet<>();
-        for (String dataDir : DatabaseDescriptor.getAllDataFileLocations())
-        {
-            for (String keyspaceDir : new File(dataDir).tryListNames())
-            {
-                // Only add a ks if it has been specified as a param, assuming params were actually provided.
-                if (keyspaceNames.length > 0 && !Arrays.asList(keyspaceNames).contains(keyspaceDir))
-                    continue;
-                keyspaces.add(keyspaceDir);
-            }
-        }
+        Set<String> keyspaces = new HashSet<>(Arrays.asList(keyspaceNames));
```

Comment:
I think you can do this now in java 11? Are we bringing this to 4.1 or 4.0 and are trying to make it java 8 compatible?
```suggestion
        Set<String> keyspaces = Set.of(keyspaceNames);
```